### ISSUE TITLE
PyTorch gradient aggregation + Horovod

### ISSUE THREAD

**Issue by aaron276h** [uber/horovod#455]
In stand-alone PyTorch it is possible to accumulate gradients from multiple batches before applying them by not calling `optimizer.step()`. However in Horovod, when we call `loss.backwards()`, we begin to send the update asynchronously in the background right away, thus not calling `optimizer.step()` does not reduce communication. What is the simplest way to emulate the stand-alone PyTorch method in Horovod?

As a follow up question, since we will be summing gradients rather than averaging them on every worker, would modifying [this line ](https://github.com/uber/horovod/blob/4f0bbbb063023f01af6c56374cf6b5ff58fa2168/horovod/torch/__init__.py#L78) be the only change needed to have Horovod sum gradietns across workers rather than average them when performing all_reduce?

**Comment by tgaddair** [comment]
Hey @aaron276h, the ability to delay aggregation past the current iteration is something we have on the roadmap, but don't currently support out of the box.

To your second question: yes, changing `average=False` will sum but not divide the gradients.
